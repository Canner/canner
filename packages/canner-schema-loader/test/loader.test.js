// import compiler from './compiler.js';
jest.setTimeout(100000);
test('Inserts schema and outputs JSONSchema and componentTree', async () => {
  // FIX: breaks at first install
  // const stats = await compiler('test.schema.js');
  // const output = stats.toJson().modules.find(module => module.name === './test.schema.js').source;
  // expect(output).toMatchSnapshot();
  // const output2 = stats.toJson().modules.find(module => module.name === './schema/string.schema.js').source;
  // expect(output2).toMatchSnapshot();
  expect(true).toBe(true);
});