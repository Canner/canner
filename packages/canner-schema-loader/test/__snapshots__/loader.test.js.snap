// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inserts schema and outputs JSONSchema and componentTree 1`] = `
"\\"use strict\\";

var _interopRequireWildcard = require(\\"@babel/runtime/helpers/interopRequireWildcard\\");

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _cannerScript = _interopRequireWildcard(require(\\"canner-script\\"));

/** @jsx builder */
var _default = (0, _cannerScript.default)(\\"root\\", null, (0, _cannerScript.default)(\\"object\\", {
  keyName: \\"test\\"
}, (0, _cannerScript.default)(\\"string\\", {
  keyName: \\"defaultUI\\",
  \\"packageName\\": \\"@canner/antd-string-input\\",
  \\"loader\\": new Promise(resolve => {
    require.ensure([], require => {
      resolve(require(\\"IGNORED_PATH/node_modules/@canner/antd-string-input/lib/index.js\\"));
    });
  }),
  \\"builder\\": null
}), (0, _cannerScript.default)(\\"string\\", {
  keyName: \\"withUI\\",
  ui: \\"textarea\\",
  \\"packageName\\": \\"@canner/antd-string-textarea\\",
  \\"loader\\": new Promise(resolve => {
    require.ensure([], require => {
      resolve(require(\\"IGNORED_PATH/node_modules/@canner/antd-string-textarea/lib/index.js\\"));
    });
  }),
  \\"builder\\": null
}), (0, _cannerScript.default)(\\"array\\", {
  keyName: \\"withBuilder\\",
  ui: \\"tag\\",
  \\"packageName\\": \\"@canner/antd-array-tag\\",
  \\"loader\\": new Promise(resolve => {
    require.ensure([], require => {
      resolve(require(\\"IGNORED_PATH/node_modules/@canner/antd-array-tag/lib/index.js\\"));
    });
  }),
  \\"builder\\": require(\\"IGNORED_PATH/node_modules/@canner/antd-array-tag/canner.def.js\\").default ? require(\\"IGNORED_PATH/node_modules/@canner/antd-array-tag/canner.def.js\\").default : require(\\"IGNORED_PATH/node_modules/@canner/antd-array-tag/canner.def.js\\")
}), (0, _cannerScript.default)(\\"array\\", {
  keyName: \\"withBuilder2\\",
  ui: \\"gallery\\",
  \\"packageName\\": \\"@canner/antd-array-gallery\\",
  \\"loader\\": new Promise(resolve => {
    require.ensure([], require => {
      resolve(require(\\"IGNORED_PATH/node_modules/@canner/antd-array-gallery/lib/index.js\\"));
    });
  }),
  \\"builder\\": require(\\"IGNORED_PATH/node_modules/@canner/antd-array-gallery/canner.def.js\\").default ? require(\\"IGNORED_PATH/node_modules/@canner/antd-array-gallery/canner.def.js\\").default : require(\\"IGNORED_PATH/node_modules/@canner/antd-array-gallery/canner.def.js\\")
}), (0, _cannerScript.default)(\\"object\\", {
  keyName: \\"withBuilder3\\",
  ui: \\"variants\\",
  \\"packageName\\": \\"@canner/antd-object-variants\\",
  \\"loader\\": new Promise(resolve => {
    require.ensure([], require => {
      resolve(require(\\"IGNORED_PATH/node_modules/@canner/antd-object-variants/lib/index.js\\"));
    });
  }),
  \\"builder\\": require(\\"IGNORED_PATH/node_modules/@canner/antd-object-variants/canner.def.js\\").default ? require(\\"IGNORED_PATH/node_modules/@canner/antd-object-variants/canner.def.js\\").default : require(\\"IGNORED_PATH/node_modules/@canner/antd-object-variants/canner.def.js\\")
}), (0, _cannerScript.default)(\\"array\\", {
  keyName: \\"withCannerConfigInPackageJson\\",
  ui: \\"table\\",
  \\"packageName\\": \\"@canner/antd-array-table\\",
  \\"loader\\": new Promise(resolve => {
    require.ensure([], require => {
      resolve(require(\\"IGNORED_PATH/node_modules/@canner/antd-array-table/lib/index.js\\"));
    });
  }),
  \\"builder\\": null,
  \\"cacheActions\\": true,
  \\"controlDeployAndResetButtons\\": true,
  \\"type\\": \\"array\\"
}), (0, _cannerScript.default)(\\"array\\", {
  keyName: \\"uiParams\\",
  ui: \\"table\\",
  uiParams: {
    columns: [{
      title: 'name',
      dataIndex: 'name'
    }]
  },
  \\"packageName\\": \\"@canner/antd-array-table\\",
  \\"loader\\": new Promise(resolve => {
    require.ensure([], require => {
      resolve(require(\\"IGNORED_PATH/node_modules/@canner/antd-array-table/lib/index.js\\"));
    });
  }),
  \\"builder\\": null,
  \\"cacheActions\\": true,
  \\"controlDeployAndResetButtons\\": true,
  \\"type\\": \\"array\\"
}, (0, _cannerScript.default)(\\"array\\", {
  keyName: \\"nested\\",
  \\"packageName\\": \\"@canner/antd-array-table\\",
  \\"loader\\": new Promise(resolve => {
    require.ensure([], require => {
      resolve(require(\\"IGNORED_PATH/node_modules/@canner/antd-array-table/lib/index.js\\"));
    });
  }),
  \\"builder\\": null,
  \\"cacheActions\\": true,
  \\"controlDeployAndResetButtons\\": true,
  \\"type\\": \\"array\\"
}, (0, _cannerScript.default)(\\"string\\", {
  keyName: \\"string\\",
  \\"packageName\\": \\"@canner/antd-string-input\\",
  \\"loader\\": new Promise(resolve => {
    require.ensure([], require => {
      resolve(require(\\"IGNORED_PATH/node_modules/@canner/antd-string-input/lib/index.js\\"));
    });
  }),
  \\"builder\\": null
}))), (0, _cannerScript.default)(\\"string\\", {
  keyName: \\"packageName\\",
  packageName: \\"./fake-string-component\\",
  \\"packageName\\": \\"./fake-string-component\\",
  \\"loader\\": new Promise(resolve => {
    require.ensure([], require => {
      resolve(require(\\"IGNORED_PATH/packages/canner-schema-loader/test/fake-string-component/index.js\\"));
    });
  }),
  \\"builder\\": require(\\"IGNORED_PATH/packages/canner-schema-loader/test/fake-string-component/canner.def.js\\").default ? require(\\"IGNORED_PATH/packages/canner-schema-loader/test/fake-string-component/canner.def.js\\").default : require(\\"IGNORED_PATH/packages/canner-schema-loader/test/fake-string-component/canner.def.js\\"),
  \\"type\\": \\"string\\",
  \\"cacheActions\\": true
})));

exports.default = _default;"
`;
